OpenSim/Simulation/Test/testForces.cpp

```diff
@@ -168,6 +176,115 @@ int main()
 //==============================================================================
 // Test Cases
 //==============================================================================
+void testSmith2016Ligament()
+{
+    using namespace SimTK;
+
+    double mass = 1;
+    double stiffness = 10;
+    double restlength = 0.5;
+    double dissipation = 0.1;
+    double start_h = 0.5;
+
+    // Setup OpenSim model
+    Model osimModel{};
+    osimModel.setName("Smith2016Ligament");
+    //OpenSim bodies
+    const Ground& ground = osimModel.getGround();;
+    OpenSim::Body pulleyBody("PulleyBody", mass ,Vec3(0), 
+                             mass*SimTK::Inertia::brick(0.1, 0.1, 0.1));
+    OpenSim::Body block("block", mass ,Vec3(0), 
+                        mass*SimTK::Inertia::brick(0.2, 0.1, 0.1));
+    block.attachGeometry(new Brick(Vec3(0.2, 0.1, 0.1)));
+    block.scale(Vec3(0.2, 0.1, 0.1), false);
+    
+    WrapCylinder* pulley = new WrapCylinder();
+    pulley->set_radius(0.1);
+    pulley->set_length(0.05);
+
+    // Add the wrap object to the body, which takes ownership of it
+    pulleyBody.addWrapObject(pulley);
+
+    // Add joints
+    WeldJoint weld("pulley", ground, Vec3(0, 1.0, 0), Vec3(0), pulleyBody, Vec3(0), Vec3(0));
+    SliderJoint slider("slider", ground, Vec3(0), Vec3(0,0,Pi/2), block, Vec3(0), Vec3(0,0,Pi/2));
+
+    double positionRange[2] = {-10, 10};
+    // Rename coordinate for the slider joint
+    auto& sliderCoord = slider.updCoordinate();
+    sliderCoord.setName("block_h");
+    sliderCoord.setRange(positionRange);
+
+    osimModel.addBody(&block);
+    osimModel.addJoint(&weld);
+
+    osimModel.addBody(&pulleyBody);
+    osimModel.addJoint(&slider);
+
+    osimModel.setGravity(gravity_vec);
+
+    Smith2016Ligament spring("spring", restlength, stiffness, dissipation);
+    spring.updGeometryPath().appendNewPathPoint("origin", block, Vec3(-0.1, 0.0 ,0.0));
+    
+    int N = 10;
+    for(int i=1; i<N; ++i){
+        double angle = i*Pi/N;
+        double x = 0.1*cos(angle);
+        double y = 0.1*sin(angle);
+        spring.updGeometryPath().appendNewPathPoint("", pulleyBody, Vec3(-x, y ,0.0));
```

Comment:
This code is "faking" wrapping by adding path points around the top half of the circle.